iceoryx_posh/test/integrationtests/test_publisher_subscriber_communication.cpp

```diff
@@ -71,6 +71,21 @@ class PublisherSubscriberCommunication_test : public RouDi_GTest
             options);
     }
 
+    // additional overload for convenient use in the test cases
+    template <typename T>
+    std::unique_ptr<iox::popo::Publisher<T>> createPublisher(const uint64_t historyCapacity)
```

Comment:
Just wondering, wouldn't this also work
```cpp
template <typename T>
iox::popo::Publisher<T>&& createPublisher(...) {
    return std::move(iox::popo::Publisher<T>(...));
}
```

Reply:
Not sure whether this would work but the current factory functions follow the style of existing ones and we should be careful not to mix to much (in tests), as it is unnecessarily confusing.